[CS2113T-T09-1] rainyDay

BenjaminPoh opened the pull request:
rainyDay is a desktop application for managing your finances. With a command line interface, you can keep track of your financial inflows and outflows and view them with a single command.

clement559 commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
```
Curly bracket comments can be omitted

YatPang replied:
This line can be used to describe the items below, rather than saying, in a literal manner, what the point of that segment is.
eg.: Listed below are the third-party sources used in this project. Points are hyperlinked to their original source.

clement559 commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+
+#### How to use this guide
+
+{todo, talk about the diagrams, formatting (markdown etc)}
```
Hopefully can see this implemented soon

YatPang replied:
Good to have a table of contents w/ each item hyperlinked to the respective headers before this section

clement559 commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+
+{insert diagrams}
+
+- stores the financial report data, such as financial statement objects
```
Can start with capitalised letter to standardise and formalise guide

YatPang replied:
{insert diagram}
Either remove this line or add diagram

clement559 commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+   the output to be shown to user
+
+### Exceptions component
+
```
Not present

clement559 commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+
+The sequence diagram for the implementation of add is as shown below:
+
+![AddCommandSequenceDiagram.png](images\DeveloperGuide\AddCommandSequenceDiagram.png)
```
Would it be easier to understand if the return result to the user is provided in explanation?

clement559 replied:
Does the final return string have an object name?

ong-ck replied:
I think the `AddCommand` object will be destroyed when its execution ends.

![image](https://user-images.githubusercontent.com/85627680/228762933-f083581e-94a0-4b75-8a29-12a92311c89c.png)

clement559 commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+
+The sequence diagram for the implementation of delete is as shown below:
+
+![DeleteCommandSequenceDiagram.png](images\DeveloperGuide\DeleteCommandSequenceDiagram.png)
```
It is unclear what the empty return lines are returning. Also, is it confusing to cast the return variable for the last step to a different type for each method?

ong-ck replied:
The return call from `FinancialReport` to `UserData` should be illustrated by a dotted line instead of a solid line.

![image](https://user-images.githubusercontent.com/85627680/228766545-f9100785-bceb-4d07-8a0a-0bb3c9aab849.png)

clement559 commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+    - Cons: user needs to know the full details of the statement and needs to type the full statement information for
+      identifying the statement to be deleted
+
+### Implementation of regex and parser
```
This section feels a bit out of place as the previous and next sections were about different commands. Would shifting this section be better?

clement559 replied:
Section is quite wordy and would be easier to understand if diagrams were provided

clement559 commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+    - This was deliberate as presenting this information at the bottom makes it easier for users to spot, 
+      as placing them at the top may cause users to miss it if they have a large table
+
+![ViewCommandSequenceDiagram.png](images\DeveloperGuide\ViewCommandSequenceDiagram.png)
```
Is the last return null necessary?

ong-ck replied:
`sortByValue` and `Convert to Valid String` should have another activation bar to represent them.

![image](https://user-images.githubusercontent.com/85627680/228767959-81404c24-4b93-463f-b7c7-99f522ca5a24.png)

ltzehan replied:
![image](https://user-images.githubusercontent.com/40513040/229009454-c2e817f6-1b73-46f5-9148-75cd01154f24.png)

Seems like the self-invocations `sortByValue` and "Convert to Valid String" are missing activation bars

clement559 commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+  much they have exceeded it by.
+- This can be seen at start-up and when the user makes any changes to their expenses for the month.
+
+![SetBudgetCommandSequenceDiagram.png](images\DeveloperGuide\SetBudgetCommandSequenceDiagram.png)
```
Does the user data activation bar start too early?

ong-ck replied:
The activation bar for `UserData` should start only after `setBudgetGoal()` has been called.

![image](https://user-images.githubusercontent.com/85627680/228763656-faa3d63a-c3e4-4a2e-ad3b-115a9d751f48.png)

ltzehan replied:
![image](https://user-images.githubusercontent.com/40513040/229009668-d9b7979a-8033-407d-860b-312e9b446a48.png)

The `:UserData` activation bar should only start after the `setBudgetGoal` arrow

clement559 commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+  then added
+  into the financial report.
+
+![EditCommand.png](images\DeveloperGuide\EditCommand.png)
```
The edit sequence is quite confusing as it requires multiple add and remove methods. Is it possible to simplify it?

clement559 commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+
+The sequence diagram for the implementation of filter is as shown below:
+
+![FilterCommand.png](images\DeveloperGuide\FilterCommand.png)
```
This sequence was very confusing as it consisted of many if-else statements. Would the empty alt blocks mean the it does not perform any functions?

clement559 reviewed (commented):
This DG was full of diagrams which was great to visualise, but some of them were quite confusing as they were quite large!

nichyjt commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+The Sequence Diagram below shows the timeline of how the different components will interact with one another, we will
+illustrate a 'delete' command, e.g. `delete 1`.
+
+![ArchitectureSequence.png](images%2FArchitectureSequence.png)
```
The sequence diagram is clean, but I'm not sure what this section of the sequence diagram is supposed to represent:

![image](https://user-images.githubusercontent.com/69704055/228727347-2a0c8c79-4db5-4186-bf58-86389e5cf313.png)

There does not seem to be a method call associated to this line and I am not sure what information this part is trying to convey. Could this be trying to represent a [constructor](https://nus-cs2113-ay2223s2.github.io/website/se-book-adapted/chapters/uml.html#object-creation)?

nichyjt replied:
Additionally, to maintain consistency in the sequence diagram, might it be better to have all solid arrows show a method?  

For example, in this part of the sequence diagram, the method name is correctly named (with the `()`):
![image](https://user-images.githubusercontent.com/69704055/228728346-6cabe9d4-f0fe-4cf4-8a76-361d9dd1033d.png)



However, this callback seems to be not following the proper syntax:
![image](https://user-images.githubusercontent.com/69704055/228727969-c80e9a13-9d98-4e62-b53e-551120fd2b65.png)

nichyjt commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+
+- stores the financial report data, such as financial statement objects
+
+- Monthly expenditures are stored in the form of a Hash Table, mapping the Year and Month to the expenditures
```
For consistency sake, perhaps it may be better to standardise all sentences to end with a fullstop?

nichyjt commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+
+- `financialReport` is achieved with the use of an ArrayList collection of the object `FinancialStatement` to model a
+  financial report containing a list of financial statements.
+- A 'FinancialStatement` object contains the following attributes:
```
There seems to be a minor typo for the backtick syntax here!

nichyjt commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+
+The sequence diagram for the implementation of add is as shown below:
+
+![AddCommandSequenceDiagram.png](images\DeveloperGuide\AddCommandSequenceDiagram.png)
```
For the sequence diagram, perhaps to better conform to the UML syntax taught in class, the bottom boxes could be removed?

This can be done by adding `hide footbox` inside the `.puml` file.

nichyjt replied:
I appreciate the addition of return types to make it easier to understand what data is being passed from different classes.

To improve on this, perhaps it may be better to show the name of the variable that is being passed back so that it gives context to the reader on what the data actually represents? 

For example:
![image](https://user-images.githubusercontent.com/69704055/228729786-45103ed6-154a-40db-bef9-d4c0c453da42.png)
The `:String` here could benefit from being renamed as 'userBudgetLimit`.

This is valid syntax accoding to the [textbook (example image linked)](https://nus-cs2113-ay2223s2.github.io/website/book/uml/sequenceDiagrams/staticMethods/images/staticMethodCall.png).

nichyjt commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+    - This was deliberate as presenting this information at the bottom makes it easier for users to spot, 
+      as placing them at the top may cause users to miss it if they have a large table
+
+![ViewCommandSequenceDiagram.png](images\DeveloperGuide\ViewCommandSequenceDiagram.png)
```
Perhaps for the method calls that does a `print` to the user, there is no need to have an arrow that points out to nothing. This is undefined syntax based on my understanding. I assume that these printing methods have a `void` signature. Might this be better if it was done as a self-invocation instead?

Sequence diagram in question:
![image](https://user-images.githubusercontent.com/69704055/228731329-39196856-e492-4f92-b9c5-5a56240b37b2.png)

Further, as mentioned earlier in the comment, it may be better to name these using your method names.

nichyjt replied:
Additionally, some of the self-invocation methods might be rendered wrongly.

For example:
![image](https://user-images.githubusercontent.com/69704055/228732280-00e553cb-d754-46bc-a052-22cb2a54ac91.png)


The callbacks do not use the activation bar syntax and are missing the return dotted error. Linked [here](https://nus-cs2113-ay2223s2.github.io/website/se-book-adapted/chapters/uml.html#self-invocation) is the syntax for self-invocation based on the textbook! 

Some of the diagrams in the DG do use the proper syntax. Might this be accidental?

nichyjt commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+| v2.0    | diligent user                       | filter my data                                               | view my past expense with ease                         |
+| v2.0    | lazy user                           | edit my entries                                              | amend my wrong entries                                 |
 
 ## Non-Functional Requirements
```
There seems to be incomplete sections for the below three sections?

nichyjt reviewed (commented):
The DG at this stage is consistently structured and contains a good wealth of information to understand how rainyDay works! 👍🏼 

I think some improvements can be made regarding sequence diagram correctness and formatting of various parts of the DG.

ong-ck commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+
+The diagram given below explains the high-level design of rainyDay, and how the components are related to one another.
+
+![architecture.png](images%2Farchitecture.png)
```
The `RainyDay` class seems to represent the `Main` class in your architecture diagram. You might want to consider renaming `Main` to `RainyDay` in your architecture diagram.

![image](https://user-images.githubusercontent.com/85627680/228761070-595ddbc4-1ec2-45c6-a50d-29a8cdb28a4e.png)

ong-ck commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+
+{todo, talk about the diagrams, formatting (markdown etc)}
+
+## Design
```
Maybe can consider adding class diagrams as well

YatPang replied:
Good to have class diagrams for clear representation

ong-ck reviewed (commented):
Great job on the DG!

ltzehan commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+
+### Architecture
+
+#### Main components of the architecture
```
Might be better for these to reflect the actual classes used in the code:
- `Main` -> `RainyDay`
- `Userdata` -> `UserData`
- `UI` -> `Ui`

HiIAmTzeKean commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+original source as well}
 
-## Design & implementation
+- [AB3](https://github.com/se-edu/addressbook-level3)
+- [OpenCSV](https://opencsv.sourceforge.net/)
+- [gson](https://github.com/google/gson)
```
Maybe you can consider a table of contents

ltzehan commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+As this is quite a long command to parse, we will use regular expressions in the following steps to match and break
+down the instructions.
+
+1. function addStatement will call function returnRemainingInformation which will check whether the input contains the
```
Consider making `addStatement` and `returnRemainingInformation` formatted as inline code so it looks clearer to the reader

HiIAmTzeKean commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+### Architecture
+
+#### Main components of the architecture
+- `Main`: Initializes the components in the correct sequence, and connects them up with each other
+- `Parser`: breaks down inputs passed from user
+- `Command`: executes the appropriate commands
+- `Userdata`: Data input by the user
+- `UI`: the user interface
+- `Storage`: reads data from and writes data to the hard disk file
```
Maybe you can consider elaborating more on these parts.

eg. UI - user interface that formats string into text output on the console...

HiIAmTzeKean commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+    - `category`: represents the type of transaction, stored as a string
+    - `date`: represents the date to be tagged of transaction, stored as LocalDate object
+
+### Adding an entry `add`
+
+- When a command is given to add a statement, the command is first parsed to check whether it follows the format of an
+  add command: `add -DIRECTION DESCRIPTION $VALUE -c CATEGORY -date DD/MM/YYYY` with the use of regex pattern
+    - Details on implementation for parsing and command fields are documented below
+- Commands in the correct format will then be parsed to extract the relevant information, and an `AddCommand` object
+  will be created with the relevant attributes
+- `RainyDay` will then call `execute` method in `Command`, where the transaction will be added into the financial report
+
+The sequence diagram for the implementation of add is as shown below:
+
```
I like how the components are broken down into chucks

HiIAmTzeKean commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+These steps will ultimately parse the user's input and extract the necessary information. If any pattern does not match,
+our parser will throw a `RainDayException` indicating the wrong input format.
+
+#### Alternatives considered
+
+We have tried using the .split() function of strings to extract information. However, as we introduced more features and
+the commands grew more complex, the .split() function became very messy and inconvenient. Using this will also result in
+us needing to do more exceptions and error handling, which just made the entire process very complicated. Thus, we opted
+to use regular expressions, which is a more tidy and logical way to parse the inputs.
+
```
The alternative method paragraph here is useful for developers who might consider going down this path!

YatPang commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,454 @@
+
+### Command component
+
+{insert diagrams}
```
line 80:
either remove the line or insert the diagram
its a greater in to leave the line there than to have no diagram (according to TA)

KN-CY committed:
Add Junit testing for ShortcutViewCommand

KN-CY committed:
Update dg

KN-CY committed:
Fix checkstyle

KN-CY committed:
Fix incorrect error message

azriellee committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113T-T09-1/tp

azriellee committed:
Merge pull request #372 from KN-CY/update-junit

Update junit

azriellee committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113T-T09-1/tp

Commit:
Refactor Parser class into a package

Commit:
Update tests of ViewCommand

Commit:
Merge branch 'CompleteBasicUI' into UG-Update

# Conflicts:
#	src/test/java/seedu/rainyDay/modules/ParserTest.java

Commit:
Update variable order declaration

Commit:
Remove unused import

Commit:
Fix view command bug for 0 remaining value

azriellee committed:
Update UG and reformatted main class slightly

azriellee committed:
Merge pull request #374 from BenjaminPoh/UG-Update

Refactor Parser Class

azriellee committed:
Amend screenshot

lil1n committed:
remove excessive information in data class diagram

lil1n committed:
fix typo

lil1n committed:
add missing words

KN-CY committed:
Merge pull request #375 from azriellee/brach-update5

Brach update5

azriellee committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113T-T09-1/tp into brach-update5

lil1n committed:
Merge branch 'master' into branch-bugFix2.0

lil1n committed:
add author tag

lil1n committed:
reorganise directory for ParserTest

KN-CY committed:
Add ppp

KN-CY committed:
Fix AboutUs links

KN-CY committed:
Rename ppp to match github username

lil1n committed:
improve Junit test for AddCommand

lil1n committed:
Merge pull request #376 from KN-CY/update-ppp

Update ppp

ChongQiRong committed:
Upgrade filter to filter by range

ChongQiRong committed:
Fix checkstyle

lil1n committed:
improve JUnit test for delete command

ChongQiRong committed:
Fix checkstyle

KN-CY committed:
Update logs to be written into logs folder

KN-CY committed:
Merge pull request #378 from ChongQiRong/filterUpgrade

Upgrade filter to filter by range

ChongQiRong committed:
Merge pull request #379 from KN-CY/update-logger

Update logs to be written into logs folder

ChongQiRong committed:
Fix bug in filter

ChongQiRong committed:
Add test for filter command parserTest for accepting a range of dates

ChongQiRong committed:
Add testing for daterange in filter command

KN-CY committed:
Rename Command name

KN-CY committed:
Move ParserTest to correct folder and add more tests

KN-CY committed:
Fix checkstyle

ChongQiRong committed:
Remove unused code

ChongQiRong committed:
Put parserTest into a package

ChongQiRong committed:
Remove unused import

KN-CY committed:
Merge pull request #380 from ChongQiRong/JunitFixFilter

Junit fix filter

KN-CY committed:
Merge branch 'temp' into update-morejunit

KN-CY committed:
Resolve merge conflicts

KN-CY committed:
Add newline

ChongQiRong committed:
Merge pull request #381 from KN-CY/update-morejunit

Update morejunit

lil1n committed:
add and update images for user guide add, delete, and view

KN-CY committed:
Standardise sequence diagrams

lil1n committed:
add checks to throw error for invalid amount

lil1n committed:
add more JUnit test for parseUserInput addCommand

KN-CY committed:
Improve storage checks

lil1n committed:
improve delete parsing

lil1n committed:
add JUnit test for delete command in parser

lil1n committed:
Merge branch 'master' into branch-bugFix2.0

# Conflicts:
#	src/test/java/seedu/rainyDay/parser/ParserTest.java

lil1n committed:
Merge pull request #383 from KN-CY/improve-storagecheck

Improve storagecheck + standardise sequence diagrams

lil1n committed:
Merge branch 'master' into branch-bugFix2.0

# Conflicts:
#	docs/DeveloperGuide.md

lil1n committed:
extract magic number

lil1n committed:
improve user guide formatting

Commit:
Fixed JunitTesting for View Command

Commit:
Remove unused import

lil1n committed:
add JUnit tests for parseUserInput add command

lil1n committed:
add imports, add empty lines and add userData

lil1n committed:
add JUnit tests for parseUserInput delete command

lil1n committed:
remove try-catch block, add throw statement

lil1n committed:
remove static from Parse Add to fix add command date bug

lil1n committed:
parser to return specific new parser object

lil1n committed:
remove static from all parser class

lil1n committed:
update user guide on max add value

lil1n committed:
update images for user guide

KN-CY committed:
Merge pull request #384 from lil1n/branch-bugFix2.0

add more JUnit tests and bug fixes

KN-CY committed:
Merge pull request #386 from BenjaminPoh/UG-Update

Fixed JunitTesting for View Command

Commit:
Update AboutUs.md

ChongQiRong committed:
Merge pull request #387 from BenjaminPoh/UG-Update

Update AboutUs.md

lil1n committed:
improve image cropping to be consistent

Commit:
Update view functionality
View now sorts by date if -sort is not included

Commit:
Update UML diagram for view command

ChongQiRong committed:
Change UG to most updated version

ChongQiRong committed:
Merge pull request #388 from BenjaminPoh/UG-Update

Update view functionality

KN-CY committed:
Merge pull request #390 from ChongQiRong/UG

Change UG to most updated version

Commit:
Fix View command bug
Removed newline at end of table

ChongQiRong committed:
Update UG

KN-CY committed:
Merge pull request #391 from BenjaminPoh/UG-Update

Fix View command bug

KN-CY committed:
Merge pull request #392 from ChongQiRong/UG2

Update UG

azriellee committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113T-T09-1/tp

azriellee committed:
Update main OOP, update test for ignore command

azriellee committed:
fix checkstyle

azriellee committed:
fix checkstyle

lil1n committed:
Merge pull request #394 from azriellee/branch-update5

Branch update5

lil1n committed:
update delete image to reflect current view sorting

lil1n committed:
update UserData class diagram

lil1n committed:
modify specific command parser to be child of parser, fix log name bug

KN-CY committed:
Update ShortcutView to standardise with other view commands

KN-CY committed:
Update test

KN-CY committed:
Merge pull request #395 from lil1n/branch-updateUserGuideImage

update user guide images

KN-CY committed:
Merge pull request #396 from lil1n/branch-improveParserOOPandFixLogName

modify specific command parser to be child of parser, fix log parser name

ChongQiRong committed:
Merge pull request #397 from KN-CY/update-shortcutview

Update ShortcutView to standardise with other view commands

KN-CY committed:
Merge pull request #398 from ChongQiRong/UGImages

Add Ug images

KN-CY committed:
Update content page + general formatting + manual testing

ChongQiRong committed:
Merge pull request #399 from KN-CY/update-dg2

Update content page + general formatting + manual testing

KN-CY committed:
Add warning and update glossary for json

KN-CY committed:
Update dg

Commit:
Update my PPP

KN-CY committed:
Update ppp

lil1n committed:
Merge pull request #400 from KN-CY/update-dg3

Update dg3

lil1n committed:
Merge pull request #401 from BenjaminPoh/UG-Update

Update my PPP

lil1n committed:
update developer guide

lil1n committed:
add 2 items to enhancement implemented and 1 item user guide contribution

lil1n committed:
Merge pull request #402 from ChongQiRong/PPP

Update PPP

lil1n committed:
remove table of content

lil1n committed:
add tp github repo version of table of content

lil1n committed:
add user guide image

lil1n committed:
replace add and delete examples with images

ChongQiRong committed:
Update DG manual testing for filter and edit commands

ChongQiRong committed:
Update table of contents

ChongQiRong committed:
Merge pull request #403 from lil1n/branch-someUpdates

updates to developer guide and ppp

ChongQiRong committed:
Merge branch 'master' into UpdateDGTesting

# Conflicts:
#	docs/DeveloperGuide.md

Commit:
Update DG (WIP)

Commit:
Merge branch 'CompleteBasicUI' into UG-Update

azriellee committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113T-T09-1/tp

azriellee committed:
Update PPP and amended typo in Qirong's PPP

Commit:
Update DG

BenjaminPoh committed:
Merge pull request #404 from ChongQiRong/UpdateDGTesting

Update DG manual testing for filter and edit commands

BenjaminPoh committed:
Merge pull request #406 from azriellee/branch-updatePPP

Branch update ppp

Commit:
Merge branch 'CompleteBasicUI' into UG-Update

# Conflicts:
#	docs/DeveloperGuide.md

Commit:
Fix Formatting

azriellee committed:
Merge pull request #407 from BenjaminPoh/UG-Update

Update DG

Commit:
Code cleanup

Commit:
Update help command

azriellee committed:
Merge pull request #408 from BenjaminPoh/UG-Update

Code cleanup

Commit:
UG Update

azriellee committed:
Merge pull request #409 from BenjaminPoh/UG-Update

UG Update

KN-CY committed:
Add rounding for storage if data changed to >2dp

lil1n committed:
Merge pull request #410 from KN-CY/improve-storagecheck2

Add rounding for storage if data changed to >2dp

Commit:
Update UG

Commit:
Merge branch 'CompleteBasicUI' into UG-Update

Commit:
Update formatting and refactoring
View command "Ignored" now aligns to table.

BenjaminPoh committed:
Merge pull request #411 from BenjaminPoh/UG-Update

Ug update

Commit:
Complete standardisation of UG
The formatting of commands were given highest priority. Note the order of explaining any command:
Preamble --> Format --> Warnings --> Notes --> Example of Usage.

Within "Example of Usage": Notes and warnings display order is not standardised to pictures. They are displayed in a logical order.

Commit:
Update DG and bug fix
Ignore/view/setbudget/help manual-testing instructions for DG added.
Fixed bug with setbudget and shortcut commands displaying wrong error messages

ChongQiRong committed:
Add DG for exceptions package

ChongQiRong committed:
Add line spacing to PPP

Commit:
Update DG TOC
removed design considerations

ChongQiRong committed:
Merge pull request #412 from BenjaminPoh/UG-Update

Complete standardisation of UG

BenjaminPoh committed:
Merge pull request #414 from ChongQiRong/DG

DG

Commit:
UG Update

ChongQiRong committed:
Merge pull request #415 from BenjaminPoh/CompleteBasicUI

UG Update

azriellee committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113T-T09-1/tp

Commit:
Add missing \n\n characters

azriellee committed:
Update DG parser diagram

Commit:
Update view command
Block timespans with 0d/w/m/y

lil1n committed:
code refactoring, extract code into methods

BenjaminPoh committed:
Merge pull request #416 from azriellee/branch-updateDG2

Branch update dg2

lil1n committed:
improve checks for add commands in parser

lil1n committed:
update author

Commit:
Fix bug in edit value

Commit:
Update DG for clarity

BenjaminPoh committed:
Merge pull request #418 from lil1n/branch-improveAddParser

improve add parser

BenjaminPoh committed:
Merge pull request #419 from ChongQiRong/editValueBug

Fix bug in edit value

azriellee committed:
Update DG diagrams

Commit:
Fix bug in editing

ChongQiRong committed:
Merge pull request #417 from BenjaminPoh/CompleteBasicUI

Update view command

Commit:
Update DG
Fix missing else in SD for View Command

BenjaminPoh committed:
Merge pull request #421 from ChongQiRong/editValueBug

Fix bug in editing

Commit:
Remove unused ArrayList class

Commit:
Update SD for ViewCommand

Commit:
Update DG

Commit:
Update filter command diagram

lil1n committed:
Merge pull request #422 from BenjaminPoh/CompleteBasicUI

Update DG

lil1n committed:
Merge pull request #423 from ChongQiRong/editValueBug

Edit value bug

KN-CY committed:
Add warning to user guide

KN-CY committed:
Rename variable to reflect saved data naming

lil1n committed:
make changes to DG

ChongQiRong committed:
Merge pull request #424 from KN-CY/update-ug2

Add warning to user guide

ChongQiRong committed:
Merge pull request #425 from KN-CY/update-comments

Rename variable to reflect saved data naming

Commit:
Fix diagrams

lil1n committed:
update dg

BenjaminPoh committed:
Merge pull request #426 from lil1n/branch-updateDG

make changes to DG

lil1n committed:
Merge pull request #427 from BenjaminPoh/update-diagrams

Fix diagrams

Commit:
PPP update

Commit:
Update AddCommmand diagram

Commit:
Add design consideration for shortcut

KN-CY committed:
Merge pull request #429 from BenjaminPoh/update-diagrams

Update AddCommmand diagram

lil1n committed:
update help image

lil1n committed:
improve help table for help command

BenjaminPoh committed:
Merge pull request #428 from ChongQiRong/PPPupdate

PPP update

Commit:
More diagram fixes

Commit:
Fix broken link

KN-CY committed:
Merge pull request #430 from BenjaminPoh/update-diagrams

More diagram fixes

lil1n committed:
update developer guide

lil1n committed:
update user guide

Commit:
Change parseFilter to throw exception when first date is after second date

KN-CY committed:
Correct user stories

KN-CY committed:
Merge pull request #431 from ChongQiRong/filterBug

Change parseFilter to throw exception when first date is after second date

KN-CY committed:
Fix broken links

lil1n committed:
Merge remote-tracking branch 'tp-repo/master' into branch-updateUserGuide2.1

# Conflicts:
#	docs/DeveloperGuide.md

lil1n committed:
Merge pull request #432 from KN-CY/fix-dg-userstories

Correct user stories

ChongQiRong committed:
Merge pull request #433 from lil1n/branch-updateUserGuide2.1

update user guide and developer guide

azriellee committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113T-T09-1/tp

# Conflicts:
#	docs/DeveloperGuide.md
#	docs/diagrams/ArchitectureSequence.puml
#	docs/images/DeveloperGuide/ArchitectureSequence.png

Commit:
Update PPP

azriellee committed:
Update DG

BenjaminPoh committed:
Merge pull request #434 from BenjaminPoh/update-diagrams

Update PPP

azriellee committed:
Update DG TOC

azriellee committed:
update DG

KN-CY committed:
Merge pull request #435 from azriellee/branch-updateDG4

Branch update dg4

azriellee committed:
Update DG

Commit:
Update Help Command

Commit:
Update help command again

azriellee committed:
Merge pull request #436 from azriellee/branch-updateDG4

Update DG

Commit:
Update UG

Commit:
Update ignorecommand phrasings

ChongQiRong committed:
Merge pull request #437 from BenjaminPoh/update-diagrams

Update help command

KN-CY committed:
Fix shortcut bug

ChongQiRong committed:
Rephrase entry to index

Commit:
Update help command again

KN-CY committed:
Round storage down instead of round off

BenjaminPoh committed:
Merge pull request #439 from ChongQiRong/UpdateUG2

Rephrase entry to index

ChongQiRong committed:
Merge pull request #440 from BenjaminPoh/update-diagrams

Update help command again

ChongQiRong committed:
Merge pull request #438 from KN-CY/fix-shortcut-bug

Fix shortcut bug

ChongQiRong committed:
Update UG

ChongQiRong committed:
Shift images to fix numbering

ChongQiRong committed:
Remove dash

KN-CY committed:
Ensure storage category has no $ character

lil1n committed:
add checks to restrict $ for category name

ChongQiRong committed:
Do not allow $ in edit

ChongQiRong committed:
Update link to glossary

lil1n committed:
Merge pull request #442 from KN-CY/storage-category-check

Ensure storage category has no $ character

ChongQiRong committed:
Remove extra \n

KN-CY committed:
Merge pull request #443 from lil1n/branch-addRestrictionToCategory

add checks to restrict $ for category name

KN-CY committed:
Merge pull request #441 from ChongQiRong/UG3

Update UG

ChongQiRong committed:
Fix error message for missing date field

ChongQiRong committed:
Fix link

lil1n committed:
add check in check description to check for $ after -c

ChongQiRong committed:
Add examples to command summary

lil1n committed:
fix JUnit

BenjaminPoh committed:
Merge pull request #444 from ChongQiRong/UG3

Fix error message for missing date field

BenjaminPoh committed:
Merge pull request #445 from lil1n/branch-lastbugfixes

add check in check description to check for $ after -c

lil1n committed:
improve $ check for category

BenjaminPoh committed:
Merge pull request #447 from lil1n/branch-furtherDescCheck

improve $ check for category

azriellee committed:
update bugs

azriellee committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113T-T09-1/tp

ChongQiRong committed:
Merge pull request #449 from azriellee/branch-bugs

Branch bugs

ChongQiRong committed:
Fix bug for edit error message if no index is given

ChongQiRong committed:
Change inline code to images and add caret for UI at startup

ChongQiRong committed:
Merge branch 'master' into UGImages

# Conflicts:
#	docs/images/UserGuide/view.png

ChongQiRong committed:
Update EXPECTED.TXT

ChongQiRong committed:
Update PPP
